src/uu/tail/src/args.rs

```diff
@@ -0,0 +1,501 @@
+//  * This file is part of the uutils coreutils package.
+//  *
+//  * For the full copyright and license information, please view the LICENSE
+//  * file that was distributed with this source code.
+
+// spell-checker:ignore (ToDO) kqueue Signum
+
+use crate::paths::Input;
+use crate::{parse, platform, Quotable};
+use clap::{Arg, ArgMatches, Command, ValueSource};
+use std::collections::VecDeque;
+use std::ffi::OsString;
+use std::time::Duration;
+use uucore::error::{UError, UResult, USimpleError, UUsageError};
+use uucore::format_usage;
+use uucore::parse_size::{parse_size, ParseSizeError};
+
+const ABOUT: &str = "\
+    Print the last 10 lines of each FILE to standard output.\n\
+    With more than one FILE, precede each with a header giving the file name.\n\
+    With no FILE, or when FILE is -, read standard input.\n\
+    \n\
+    Mandatory arguments to long flags are mandatory for short flags too.\
+    ";
+const USAGE: &str = "{} [FLAG]... [FILE]...";
+
+pub mod options {
+    pub mod verbosity {
+        pub static QUIET: &str = "quiet";
+        pub static VERBOSE: &str = "verbose";
+    }
+    pub static BYTES: &str = "bytes";
+    pub static FOLLOW: &str = "follow";
+    pub static LINES: &str = "lines";
+    pub static PID: &str = "pid";
+    pub static SLEEP_INT: &str = "sleep-interval";
+    pub static ZERO_TERM: &str = "zero-terminated";
+    pub static DISABLE_INOTIFY_TERM: &str = "-disable-inotify"; // NOTE: three hyphens is correct
+    pub static USE_POLLING: &str = "use-polling";
+    pub static RETRY: &str = "retry";
+    pub static FOLLOW_RETRY: &str = "F";
+    pub static MAX_UNCHANGED_STATS: &str = "max-unchanged-stats";
+    pub static ARG_FILES: &str = "files";
+    pub static PRESUME_INPUT_PIPE: &str = "-presume-input-pipe"; // NOTE: three hyphens is correct
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum Signum {
+    Negative(u64),
+    Positive(u64),
+    PlusZero,
+    MinusZero,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum FilterMode {
+    Bytes(Signum),
+
+    /// Mode for lines delimited by delimiter as u8
+    Lines(Signum, u8),
+}
+
+impl FilterMode {
+    fn from(matches: &ArgMatches) -> UResult<Self> {
+        let zero_term = matches.contains_id(options::ZERO_TERM);
+        let mode = if let Some(arg) = matches.value_of(options::BYTES) {
+            match parse_num(arg) {
```

Comment:
Can we make `parse_num` return `UResult<Signum>`?

Reply:
Yep, I was a little bit torn here between keeping the function general or optimizing it. But, I'm on your side and also tend to the latter, here.